cfme/tests/perf/workloads/test_memory_leak.py

```diff
@@ -0,0 +1,103 @@
+import time
+import pytest
+
+from cfme.utils import conf, testgen
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.grafana import get_scenario_dashboard_urls
+from cfme.utils.log import logger
+from cfme.utils.providers import ProviderFilter
+from cfme.utils.smem_memory_monitor import add_workload_quantifiers, SmemMemoryMonitor
+from cfme.utils.workloads import get_memory_leak_scenarios
+
+
+roles_memory_leak = ['automate', 'database_operations', 'ems_inventory', 'ems_metrics_collector',
+    'ems_metrics_coordinator', 'ems_metrics_processor', 'ems_operations', 'event', 'notifier',
+    'reporting', 'scheduler', 'user_interface', 'web_services']
+
+pytest_generate_tests = testgen.generate(
+    gen_func=testgen.providers,
+    filters=[ProviderFilter()],
+    scope="module"
+)
+
+
+def prepare_workers(appliance):
+    """Set single instance of each worker type and maximum threshold"""
+    view = navigate_to(appliance.server, 'Workers')
+    view.workers.fill({
+        "generic_worker_count": "1",
+        "cu_data_collector_worker_count": "1",
+        "ui_worker_count": "1",
+        "reporting_worker_count": "1",
+        "web_service_worker_count": "1",
+        "priority_worker_count": "1",
+        "cu_data_processor_worker_count": "1",
+        "vm_analysis_collectors_worker_count": "1",
+        "websocket_worker_count": "1",
+        "generic_worker_threshold": "1.5 GB",
+        "cu_data_collector_worker_threshold": "1.5 GB",
+        "event_monitor_worker_threshold": "10 GB",
+        "connection_broker_worker_threshold": "10 GB",
+        "reporting_worker_threshold": "1.5 GB",
+        "web_service_worker_threshold": "1.5 GB",
+        "priority_worker_threshold": "1.5 GB",
+        "cu_data_processor_worker_threshold": "1.5 GB",
+        "refresh_worker_threshold": "10 GB",
+        "vm_analysis_collectors_worker_threshold": "1.5 GB"
+    })
+    view.workers.save.click()
+
+
+@pytest.mark.usefixtures('generate_version_files')
+@pytest.mark.parametrize('scenario', get_memory_leak_scenarios())
+def test_workload_memory_leak(request, scenario, appliance, provider):
+    """Runs through provider based scenarios setting one worker instance and maximum threshold and
+    running for a set period of time. Memory Monitor creates graphs and summary info."""
+    from_ts = int(time.time() * 1000)
```

Comment:
**QUESTION:** We are taking the time in seconds and multiplying it by 1000......But then further down doing things with 300.....

Reply:
`from_ts` is used to create reports after test, further down part of the code refers to how long the test should run

Reply:
but if ```time.time()``` gives you number of seconds......*1000 turns it into milliseconds right?

Reply:
Yep, it used only in `get_scenario_dashboard_urls(scenario, from_ts, to_ts)`  to build data to grafana